jax/experimental/djax.py

```diff
@@ -954,9 +955,10 @@ def new_res(v):
     unks = map(read, eqn.invars)
     if any(unks):
       invars = [v if unk else new_res(v) for unk, v in zip(unks, eqn.invars)]
-      eqns2.append(pe.new_jaxpr_eqn(invars, eqn.outvars, eqn.primitive,
-                                    eqn.params,
-                                    source_info_util.new_source_info()))
+      eqns2.append(core.new_jaxpr_eqn(invars, eqn.outvars, eqn.primitive,
+                                      eqn.params,
+                                      None,
```

Comment:
```suggestion
                                      set(),
```

Reply:
An alternative to writing `set()` literals everywhere would be to define a `no_effects` variable and use that.

Also maybe join the lines here.

Reply:
Done.